app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -244,19 +248,80 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote, val w
 	 * not authorized then the the [QueueMetadata] is created from the app remote API.
 	 */
 	fun createLikedSongsQueueMetadata() {
+		if (createQueueMetadataJob?.isActive == true) {
+			createQueueMetadataJob?.cancel()
+		}
+
 		createQueueMetadataJob = GlobalScope.launch(defaultDispatcher) {
 			queueItems = webApi.getLikedSongs(this@SpotifyAppController) ?: emptyList()
+
 			if (queueItems.isNotEmpty()) {
-				queueMetadata = QueueMetadata("Liked Songs", null, queueItems)
-				loadQueueCoverart()
+				queueMetadata = QueueMetadata(L.MUSIC_LIKED_SONGS_PLAYLIST_NAME, null, queueItems)
```

Comment:
Good catch, that this should've been translated before, but now that I'm seeing it, why not just use the Spotify-provided name for this playlist that we are overriding? Perhaps save the actual playerContext queueTitle at the top with queueUri and friends, and refer back to it as needed instead of providing our own?

Reply:
Ah that's a much better approach than what I was doing here, then we don't have to worry about translations and all that! Added a queueTitle member var and updated it with the playerContext from the callback